seed/ts-sdk/oauth-client-credentials-environment-variables/token-override/src/auth/OAuthAuthProvider.ts

```diff
@@ -0,0 +1,149 @@
+                if (clientId == null) {
+                    throw new errors.SeedOauthClientCredentialsEnvironmentVariablesError({
+                        message:
+                            "clientId is required; either pass it as an argument or set the CLIENT_ID environment variable",
+                    });
+                }
+
+                const clientSecret = (await core.Supplier.get(this._clientSecret)) ?? process.env?.CLIENT_SECRET;
+                if (clientSecret == null) {
+                    throw new errors.SeedOauthClientCredentialsEnvironmentVariablesError({
+                        message:
+                            "clientSecret is required; either pass it as an argument or set the CLIENT_SECRET environment variable",
+                    });
+                }
+                const tokenResponse = await this._authClient.getTokenWithClientCredentials({
+                    client_id: clientId,
+                    client_secret: clientSecret,
+                });
+
+                this._accessToken = tokenResponse.access_token;
+                this._expiresAt = this.getExpiresAt(tokenResponse.expires_in, this.BUFFER_IN_MINUTES);
+                return this._accessToken;
+            } finally {
+                this._refreshPromise = undefined;
+            }
+        })();
+        return this._refreshPromise;
+    }
+
+    private getExpiresAt(expiresInSeconds: number, bufferInMinutes: number): Date {
+        const now = new Date();
+        return new Date(now.getTime() + expiresInSeconds * 1000 - bufferInMinutes * 60 * 1000);
+    }
+}
+
+export class OAuthTokenOverrideAuthProvider implements core.AuthProvider {
+    private readonly _token: core.Supplier<string>;
+
+    constructor(options: OAuthAuthProvider.Options & OAuthAuthProvider.AuthOptions.TokenOverride) {
+        if (options.token == null) {
+            throw new errors.SeedOauthClientCredentialsEnvironmentVariablesError({
+                message: "token is required. Please provide it in options.",
+            });
+        }
+        this._token = options.token;
+    }
+
+    public static canCreate(
+        options: OAuthAuthProvider.Options,
+    ): options is OAuthAuthProvider.Options & OAuthAuthProvider.AuthOptions.TokenOverride {
+        return "token" in options && options.token != null;
+    }
+
+    public async getAuthRequest(_arg?: { endpointMetadata?: core.EndpointMetadata }): Promise<core.AuthRequest> {
+        return {
+            headers: {
+                Authorization: `Bearer ${await await core.Supplier.get(this._token)}`,
+            },
+        };
+    }
+}
+
+export namespace OAuthAuthProvider {
+    export type AuthOptions = AuthOptions.ClientCredentials | AuthOptions.TokenOverride;
+
+    export namespace AuthOptions {
+        export interface ClientCredentials {
+            clientId: core.Supplier<string>;
+            clientSecret: core.Supplier<string>;
+        }
```

Comment:
This should be:
```ts
export interface ClientCredentials {
    clientId: core.Supplier<string> | undefined;
    clientSecret: core.Supplier<string> | undefined;
}
```
because it should fallback to fetching from process.env